arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1741,87 +1741,49 @@ void RestReplicationHandler::handleCommandRestoreView() {
   LOG_TOPIC(TRACE, Logger::REPLICATION) << "restoring view: "
     << nameSlice.copyString();
 
-  if (ServerState::instance()->isCoordinator()) {
-    try {
-      auto* ci = ClusterInfo::instance();
-      auto view = ci->getView(_vocbase.name(), nameSlice.toString());
-
-      if (view) {
-        if (overwrite) {
-          auto res = view->drop();
+  try {
+    CollectionNameResolver resolver(_vocbase);
+    auto view = resolver.getView(nameSlice.toString());
 
-          if (!res.ok()) {
-            generateError(res);
+    if (view) {
+      if (overwrite) {
+        auto res = view->drop();
 
-            return;
-          }
-
-          generateError(TRI_ERROR_ARANGO_DUPLICATE_NAME);
+        if (!res.ok()) {
+          generateError(res);
 
           return;
         }
-      }
 
-      auto res = LogicalView::create(view, _vocbase, slice); // must create() since view was drop()ed
-
-      if (!res.ok()) {
-        generateError(res);
-
-        return;
-      }
-
-      if (!view) {
-        generateError(Result(TRI_ERROR_INTERNAL, "problem creating view"));
+        generateError(TRI_ERROR_ARANGO_DUPLICATE_NAME);
```

Comment:
I think this is wrong, the original code had an else branch here. If the _overwrite_ flag is set you need to drop **and** recreate the view.

Reply:
good catch, I wonder why no tests caught it